lib/travis/api/v3/services/beta_migration_request/create.rb

```diff
@@ -0,0 +1,37 @@
+module Travis::API::V3
+  class Services::BetaMigrationRequest::Create < Service
+    params :organizations
+    result_type :beta_migration_request
+
+    def run!
+      current_user = check_login_and_find(:user)
+      not_found(false, :beta_migration_request) unless current_user
+
+      organizations = validate_organizations(current_user)
+
+      beta_migration_request = query(:beta_migration_request).create(current_user, organizations)
+
+      beta_migration_request.save!
+      enable_beta(organizations + [current_user])
+
+      result beta_migration_request
+    end
+
+    def enable_beta(owners)
+      owners.each do |org|
+        Travis::Features.activate_owner(:beta_migration_opt_in, org)
+      end
+    end
+
+    def validate_organizations(current_user)
```

Comment:
I think it might be better to run a single query here to avoid N+1 queries problem. It won't probably matter that much because I don't think that anyone in the DB has more than 200 orgs, but I think it might also make for a more readable code, something like:

```ruby
def valid_organizations(current_user)
  Models::Organization.where(id: params['organizations'], memberships: {user_id: current_user.id, role: "admin"}).joins(:memberships)
end
```

Reply:
That's good, fixed